### DISCUSSION TITLE
Discord Community Support?

### DISCUSSION THREAD

**Discussion by RiemaruKarurosu** [noisetorch/NoiseTorch#305] [Ideas (I would suggest ...)]
It would be nice to create a discord server focused on basic linux users Issue Support (and add it to README).

Note: 
I don't need it, but I saw them in some FOSS apps (Like lutris) to create a community.

**Comment by Technetium1** [comment]
I think a general Linux troubleshooting server might be a bit out of scope, but ZyanKLeee will be the decider. I'd like to refer you to this for context as to why Discord may not be the best format: https://kotaku.com/please-stop-closing-forums-and-moving-people-to-discord-1847684851

**Comment by ZyanKLee** [comment]
Hey there,

I noticed the tendency for projects to move their communities to discord as well (OBS, PhantomBot, Stream-PI, and several others). Though I'm not sure it did them really any good. Discord has some shortcomings over forums (i.e. github discussions) which are mostly explained in the link @Technetium1 already posted. For me personally the most important one is:

Discord is a real-time communication plattform - in text chats as well as the voice channels. Which is fine for real-time or at least near-time communication.

They therefore either require constant monitoring (which I can and will not do - and also won't expect from anybody else) or questions/topics will be overlooked and not answered.

We got [discussions](/noisetorch/NoiseTorch/discussions) set up, to cater to that need of communication which does not directly relate to a shortcoming of NoiseTorch or its development workflow itself. That said, I will move this topic to discussions 😉 

Don't get me wrong: I do like Discord - it does the real-time communication really well and when gaming with my friends, this is our app to go to for voice chat.

**Reply by ZyanKLee** [reply]
If enough people would like some real-time communication platform in addition to the asynchronous discussions, I would rather use something like Matrix, Jabber/XMPP, Mattermost, IRC or even Slack or Gitter.

Does anyone see a need?

**Reply by TheDukeofErl** [reply]
The biggest advantage I see in having some realtime chat available is in the case of dev to dev communications, for both regular contributors as well as those who are new contributors. Sometimes there are questions someone might want to ask that doesn't necessarily fit an issue/discussion. If this were done, it would have to be something that doesn't become a support channel as well as something that isn't just constantly random conversations. That could lead to a whole slew of moderation issues, which I doubt any of us want to deal with. 

I agree with the sentiment @Technetium1 has in regards to Discord and how it may not be the best format. "Discordification" can cause issues with the ability to find information without having to actually join the community and then lead to more confusion when actually trying to search for information. 

I'm not sure what things look like in the post-freenode era, I kind of stopped paying attention to it, as I was never really active on FOSS IRC channels to begin with, but I imagine most FOSS projects still use IRC (libera.chat?) or have switched to Matrix for that purpose.

**Reply by ZyanKLee** [reply]
Another point that just jumped my mind:

we are quite distributed around the world. I'm from central Europe and know at least of one person being from Canada - which is a difference of at least 6 hours.

Finding a timeslot for synchronous communication may be difficult.

So, the sooner we find a way to organize with asynchronous communication, the better it is for all of us.

Being patient is key to this. Until everybody even had a chance to be awake and look at something, 24 hours may have passed.

Discussing multiple topics in multiple parallel threads may be a time saver here.
So don't hold back: ask away, suggest the bloody hell out of the discussions and create PRs as many in parallel as you like.
Producing and resolving conflicts between branches is not a crime, either, but a necessity.

**Reply by AXDOOMER** [reply]
I may not be adding much here, but here's my opinion: 

I don't see the need for a Discord or other real-time communication. When you look at communities (such as Lutris, OBS, PhantomBot and Stream-PI mentioned here) that have a Discord or similar chat for their community, it's because they are highly customizable software. They are complex software and lots of things could go wrong when using them because of the amount of features they have. Having such a community is great because the developers don't have the time to help everyone who has issues with the software so other people in the community will help those people troubleshoot their issues instead. 

I think it's different for NoiseTorch because the software is very minimalist. NoiseTorch has four different settings, it's highly beginner friendly and if something doesn't work, it's most likely a bug. I think so far the Issues have been welcoming to any types of questions and users shall always feel welcome to ask them, even if they feel their question is "stupid".

**Reply by Technetium1** [reply]
@ZyanKLee 
> If enough people would like some real-time communication platform in addition to the asynchronous discussions, I would rather use something like Matrix, Jabber/XMPP, Mattermost, IRC or even Slack or Gitter.

Of all of those choices, [Matrix](https://matrix.org/discover/) absolutely seems like the best one by far. With IRC in second, but there's not any scrollback with IRC, which **really sucks** for people wanting to search for recent issues and therefore wouldn't suit the community's needs. You'd also have to roll your own web interface to some extent to make it usable.

Gitter has a scrollback and does not require an account to start looking, and Gitter allows OAuth login via GitLab, Twitter, and GitHub. Another point for Matrix as well, because they were [acquired](https://blog.gitter.im/2020/09/30/gitter-element-acquisition/) by [Element](https://element.io), and therefore have started running the backbone on Matrix.

[Rocket.Chat](https://rocket.chat) might also become a possibility since 2 days ago they've also [opened up to Matrix](https://matrix.org/blog/2022/05/30/welcoming-rocket-chat-to-matrix). Matrix also offers true end-to-end encryption, which IRC cannot offer, and which cannot be achieved over XMPP without plugins and excessive messing about.

> Finding a timeslot for synchronous communication may be difficult.

That can be worked out, there are tons of [free] products for that, *BUT* it can also be done via bots; of which there are many.


---

@TheDukeofErl 
> The biggest advantage I see in having some realtime chat available is in the case of dev to dev communications

I agree with this point, it's easier to either verbalize in a common language than it can be to post here.

> "Discordification" can cause issues with the ability to find information without having to actually join the community and then lead to more confusion when actually trying to search for information

This is absolutely true and part of the reason there would have to be scrollback with a search function.

> I'm not sure what things look like in the post-freenode era

Well... They kinda suck unless you're self-hosting. Most from there have moved to [Libera](https://libera.chat/) since the new 'owner' of Freenode is a monster. My thoughts against IRC are above though.
